src/vaults/BalanceSheetManager.sol

```diff
@@ -219,6 +219,27 @@ contract BalanceSheetManager is
         _revoke(poolId, scId, from, pricePerShare, shares);
     }
 
+    /// @inheritdoc IBalanceSheetManagerGatewayHandler
+    function approvedDeposits(PoolId poolId, ShareClassId scId, AssetId assetId, uint128 assetAmount) external auth {
+        (address token, uint256 tokenId) = poolManager.checkedIdToAsset(assetId.raw());
+        // TODO(wischli): Rm hardcode post #187 merge
+        D18 pricePoolPerAsset = d18(1);
+        JournalEntry[] memory journalEntries = new JournalEntry[](0);
+        Meta memory meta = Meta(journalEntries, journalEntries);
+
+        escrow.pendingDepositIncrease(token, tokenId, poolId.raw(), scId.raw(), assetAmount);
+        sender.sendUpdateHoldingAmount(
+            poolId, scId, assetId, address(escrow), assetAmount, pricePoolPerAsset, true, meta
+        );
+        this.updateValue(poolId, scId, token, tokenId, pricePoolPerAsset);
+    }
+
+    /// @inheritdoc IBalanceSheetManagerGatewayHandler
+    function revokedShares(PoolId poolId, ShareClassId scId, AssetId assetId, uint128 assetAmount) external auth {
+        (address token, uint256 tokenId) = poolManager.checkedIdToAsset(assetId.raw());
+        escrow.reserveIncrease(token, tokenId, poolId.raw(), scId.raw(), assetAmount);
```

Comment:
NOTE: 

* For new approach: need to find a substitute for reserve action

Reply:
Missing: 

* `AsyncInvestmentManager.redeem` to reduce reservance on perPoolEscrow

Reply:
> NOTE:
> 
> * For new approach: need to find a substitute for reserve action

@mustermeiszer Can you elaborate on that? It seems like it requires a code change for this PR and I don't fully know what/how.

Reply:
> Missing:
> 
> * `AsyncInvestmentManager.redeem` to reduce reservance on perPoolEscrow

Just to clarify: IIUC, `AsyncRequests.redeem` needs to do two things:
1. `escrow.reserveDecrease` (reverting increase from `bsm.revokedShares`)
2. `bsm.withdraw` (which decreases the deposit amount that is increased as a result of `bsm.approvedDeposits`)